File: Chess.Angular/ClientApp/src/app/chess/models/boardCell.ts
import { X_axic, Y_axic, CellColor } from "./cellPropeties";
import { ChessUnit } from "./chessUnit";

export class BoardCell{
    readonly Color: CellColor;
    
    Unit: ChessUnit | null;
    X: X_axic;
    Y: Y_axic;

    isPossibleMove: boolean;
    isSelected: boolean;

    constructor(color: CellColor, unit: ChessUnit | null, x: X_axic, y: Y_axic){
        this.Color = color;
        this.Unit = unit;
        this.X = x;
        this.Y = y;
        this.isPossibleMove = false;
        this.isSelected = false;
    }


}

File: Chess.Domain/Models/BoardCell.cs
using Chess.Domain.Enums;

namespace Chess.Domain.Models
{
    public class BoardCell
    {
        public BoardCell(CellColor cellColor, X_axic x, Y_axic y, ChessUnit? unit)
        {
            CellColor = cellColor;
            X = x;
            Y = y;
            Unit = unit;

            IsPossibleMove = false;
            IsSelected = false;
        }

        public CellColor CellColor { get; }

        public X_axic X { get; }

        public Y_axic Y { get; }

        public ChessUnit? Unit { get; set; }

        public bool IsPossibleMove { get; set; }

        public bool IsSelected { get; set; }
    }
}
